djpsa/halo/utils.py

```diff
@@ -1,43 +1,100 @@
+import time
 import redis
 import logging
 import requests
-
+from contextlib import contextmanager
 
 from django.conf import settings
 
 logger = logging.getLogger(__name__)
 
+ONE_HOUR = 3600
+REQUEST_TIMEOUT = 30
+LOCK_TIMEOUT = REQUEST_TIMEOUT + 1
+BLOCK_TIMEOUT = 60
 
-redis_client = redis.StrictRedis(
-        host=settings.REDIS['host'],
-        port=settings.REDIS['port'],
-        password=settings.REDIS['password'],
-    )
+redis_client = None
+
+
+def get_redis_client():
+    global redis_client
+
+    if not redis_client:
+        redis_client = redis.StrictRedis(
+            host=settings.REDIS['host'],
+            port=settings.REDIS['port'],
+            password=settings.REDIS['password'],
+        )
+
+    return redis_client
+
+
+class LockNotAcquiredError(Exception):
+    """Custom exception raised when the lock is not acquired."""
+    pass
+
+
+@contextmanager
+def redis_lock(lock_name):
```

Comment:
FWIW since it exists, it can stay, but if there's only going to be one use of a context manager, it's probably not time yet to write the context manager.